src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -817,15 +813,15 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
             }
         }
 
-        // Ensure the `IsNotConst` qualification is preserved.
+        // Ensure the `IsNotPromotable` qualification is preserved.
         // NOTE(eddyb) this is actually unnecessary right now, as
         // we never replace the local's qualif, but we might in
         // the future, and so it serves to catch changes that unset
         // important bits (in which case, asserting `contains` could
         // be replaced with calling `insert` to re-set the bit).
         if kind == LocalKind::Temp {
```

Comment:
Is this check good/needed? Or should all kinds of locals have this treatment?

Reply:
good call, this can be for all locals now